### PR TITLE
LLVM and SPIRV-LLVM-Translator pulldown (WW50 2024)

### PR DESCRIPTION
**Pull request by iclsrc**
LLVM: llvm/llvm-project@53c0a25db7a0469f6d47e130d5a0e8f7a88b9585
SPIRV-LLVM-Translator: KhronosGroup/SPIRV-LLVM-Translator@b8b1b96bd7419c1561089e89dbee4b8681634ed2

### CODE REVIEW

**Comment by jsji** [comment]
This is ready for review.

- [ ] [[SYCL][Test] Update test after](https://github.com/intel/llvm/pull/16372/commits/70a78175feb7ff1bfb75341cf521450a0a71cb91) https://github.com/intel/llvm/commit/4d6e69143dc449814884ac649583d3b35bc4ae91 [(#25932)](https://github.com/intel/llvm/pull/16372/commits/70a78175feb7ff1bfb75341cf521450a0a71cb91)

- [ ]  [[Clang] Allow intel_reqd_sub_group_size attr for CUDA SPIRV (#25953)](https://github.com/intel/llvm/pull/16372/commits/8e9f1e3607744d5dd33794e31434c407d5faa466) 
- [ ] [Fix fp-accuracy for atan2 after](https://github.com/intel/llvm/pull/16372/commits/39bf43b9f17975d83c7ece3f23dadea25db1d704) https://github.com/intel/llvm/commit/5c2a133b1342881dc4f42a896e7e5f4b85d20508 
- [ ] [[SYCL] Add initializes attr to vector_math_ops.cpp after](https://github.com/intel/llvm/pull/16372/commits/877f31225b1699d1ed38a6b06d1f2e309549c955) https://github.com/intel/llvm/commit/4d6e69143dc449814884ac649583d3b35bc4ae91

- [ ] [[SPIRV] Disable RUN lines for joint matrix failures](https://github.com/intel/llvm/pull/16372/commits/73f8fbf93d65fe29396b7334948efc4bf815e5e5)
- [ ] [[Clang][SemaCUDA] Sync test with community for amdgpu-attrs.cu](https://github.com/intel/llvm/pull/16372/commits/a6dc6bedb0d3793d94367fce9845e12062d35d83)
- [ ] [AMDHIP: Add -Wno-error=deprecated-pragma for __AMDGCN_WAVEFRONT_SIZE](https://github.com/intel/llvm/pull/16372/commits/eb615f582b054d8d959a53defca9e73f90ce46be)

File: sycl/test-e2e/EnqueueNativeCommand/custom-command-hip.cpp [hunk 1]
```diff
@@ -1,6 +1,6 @@
 // FIXME: the rocm include path and link path are highly platform dependent,
 // we should set this with some variable instead.
-// RUN: %{build} -o %t.out -I%rocm_path/include -L%rocm_path/lib -lamdhip64
+// RUN: %{build} -Wno-error=deprecated-pragma -o %t.out -I%rocm_path/include -L%rocm_path/lib -lamdhip64
```
Comment (Fznamznon) [inline sycl/test-e2e/EnqueueNativeCommand/custom-command-hip.cpp]: I wonder why "deprecated-pragma" is passed, when __AMDGCN_WAVEFRONT_SIZE is a macro?

Comment (jsji) [reply sycl/test-e2e/EnqueueNativeCommand/custom-command-hip.cpp]: Good question, the error message was `-Werror, -Wdeprecated-pragma` in test. Not sure which clang change caused that.

Comment (jsji) [reply sycl/test-e2e/EnqueueNativeCommand/custom-command-hip.cpp]: Sure. https://github.com/intel/llvm/issues/16395 to track.

**Review by Fznamznon — APPROVED** [review]

File: clang/lib/CodeGen/CGBuiltin.cpp [hunk 2]
```diff
@@ -2979,8 +2979,8 @@ RValue CodeGenFunction::EmitBuiltinExpr(const GlobalDecl GD, unsigned BuiltinID,
     case Builtin::BI__builtin_atan2l:
     case Builtin::BI__builtin_atan2f128:
       return RValue::get(emitBinaryMaybeConstrainedFPBuiltin(
-          *this, E, Intrinsic::atan2,
-          Intrinsic::experimental_constrained_atan2));
+          *this, E, Intrinsic::atan2, Intrinsic::experimental_constrained_atan2,
```
Comment (Fznamznon) [inline clang/lib/CodeGen/CGBuiltin.cpp]: @zahiraam , could you please take a look?

Comment (zahiraam) [reply clang/lib/CodeGen/CGBuiltin.cpp]: @Fznamznon I think I had already reviewed this and approved it.

Comment (Fznamznon) [reply clang/lib/CodeGen/CGBuiltin.cpp]: thanks!

Comment (jsji) [reply clang/lib/CodeGen/CGBuiltin.cpp]: This is cherry-pick of @zahiraam 's fix. :)

**Review by vmaksimo — APPROVED** [review]
approve for llvm-spirv changes

**Comment by jsji** [comment]
@intel/llvm-gatekeepers This is ready for merge. Can someone help to issue a /merge. Thanks.

**Review by bb-sycl — APPROVED** [review]

**Comment by bb-sycl** [comment]
Tue 17 Dec 2024 06:27:34 PM UTC --- Merge failed with error: Please check whether the PR is mergeable

**Comment by sarnex** [comment]
@jsjj Git version issue again?

**Review by bb-sycl — APPROVED** [review]

**Comment by bb-sycl** [comment]
Tue 17 Dec 2024 06:41:05 PM UTC --- Start to merge the commit into sycl branch. It will take several minutes.

**Comment by bb-sycl** [comment]
Tue 17 Dec 2024 06:41:06 PM UTC --- Merge failed with error: Please check whether the PR is mergeable

**Review by bb-sycl — APPROVED** [review]

**Comment by bb-sycl** [comment]
Tue 17 Dec 2024 06:47:17 PM UTC --- Start to merge the commit into sycl branch. It will take several minutes.

**Comment by bb-sycl** [comment]
Tue 17 Dec 2024 06:47:19 PM UTC --- Merge failed with error: Please check whether the PR is mergeable

**Comment by jsji** [comment]
@DoyleLi The bot is still not working, please have a look. Thanks.

**Review by bb-sycl — APPROVED** [review]

**Comment by bb-sycl** [comment]
Wed 18 Dec 2024 02:03:33 AM UTC --- Merge failed with error: Please check whether the PR is mergeable

**Review by bb-sycl — APPROVED** [review]

**Comment by bb-sycl** [comment]
Wed 18 Dec 2024 02:19:50 AM UTC --- Start to merge the commit into sycl branch. It will take several minutes.

**Comment by bb-sycl** [comment]
Wed 18 Dec 2024 02:25:13 AM UTC --- Merge the branch in this PR to base automatically. Will close the PR later.

**Comment by DoyleLi** [comment]
> @DoyleLi The bot is still not working, please have a look. Thanks.

Hi @jsji the issue is resolved and will not happen in the future. I retriggered the job and verified the whole process.

**Comment by jsji** [comment]
> > @DoyleLi The bot is still not working, please have a look. Thanks.
> 
> Hi @jsji the issue is resolved and will not happen in the future. I retriggered the job and verified the whole process.

Thanks @DoyleLi

**Pull request merged**